fl4health/model_bases/masked_layers.py

```diff
@@ -50,12 +21,35 @@ def __init__(
         device: Optional[torch.device] = None,
         dtype: Optional[torch.dtype] = None,
     ) -> None:
+        """
+        Implementation of masked linear layers.
+
+        Like regular linear layers (i.e., nn.Linear module), a masked linear layer has a weight and a bias.
+        However, the weight and the bias do not receive gradient in back propagation.
+        Instead, two score tensors - one for the weight and another for the bias - are maintained.
+        In the forward pass, the score tensors are transformed by the Sigmoid function into probability scores,
+        which are then used to produce binary masks via bernoulli sampling.
+        Finally, the binary masks are applied to the weight and the bias. During training,
+        gradients with respect to the score tensors are computed and used to update the score tensors.
+
+        Note: the scores are not assumed to be bounded between 0 and 1.
+
+        Args:
+            in_features: size of each input sample
+            out_features: size of each output sample
+            bias: If set to ``False``, the layer will not learn an additive bias.
+                Default: ``True``
+
+        Attributes:
+            weight: weights of the module.
+            bias:  bias of the module.
+            weight_score: learnable scores for the weights. Has the same shape as weight.
+            bias_score: learnable scores for the bias. Has the same shape as bias.
+        """
         super().__init__(in_features, out_features, bias, device, dtype)
         self.in_features = in_features
         self.out_features = out_features
-        self.weight = Parameter(
-            torch.empty((out_features, in_features), device=device, dtype=dtype), requires_grad=False
-        )
+        self.weight.requires_grad = False
         self.weight_scores = Parameter(torch.randn_like(self.weight), requires_grad=True)
         if bias:
             self.bias = Parameter(torch.empty(out_features, device=device, dtype=dtype), requires_grad=False)
```

Comment:
Since we're dropping the empyy for weight and just setting `weight.requires_grad = False`, we should probably do that here too yes?

Reply:
Yes you are right.